src/main/java/me/THEREALWWEFAN231/tunnelmc/translator/packet/world/LevelChunkDecoder.java

```diff
@@ -0,0 +1,121 @@
+            byte paletteHeader = byteBuf.readByte();
+            boolean isRuntime = (paletteHeader & 1) == 1;
+            int paletteVersion = (paletteHeader | 1) >> 1;
+
+            BitArrayVersion bitArrayVersion = BitArrayVersion.get(paletteVersion, true);
+
+            int maxBlocksInSection = 4096;
+            BitArray bitArray = bitArrayVersion.createPalette(maxBlocksInSection);
+            int wordsSize = bitArrayVersion.getWordsForSize(maxBlocksInSection);
+
+            for (int wordIterationIndex = 0; wordIterationIndex < wordsSize; wordIterationIndex++) {
+                int word = byteBuf.readIntLE();
+                bitArray.getWords()[wordIterationIndex] = word;
+            }
+
+            int paletteSize = VarInts.readInt(byteBuf);
+            int[] sectionPalette = new int[paletteSize];
+            NBTInputStream nbtStream = isRuntime ? null : new NBTInputStream(new NetworkDataInputStream(new ByteBufInputStream(byteBuf)));
+            for (int i = 0; i < paletteSize; i++) {
+                if (isRuntime) {
+                    sectionPalette[i] = VarInts.readInt(byteBuf);
+                } else {
+                    try {
+                        NbtMapBuilder map = ((NbtMap) nbtStream.readTag()).toBuilder();
+                        map.replace("name", "minecraft:" + map.get("name").toString());
+
+                        sectionPalette[i] = BlockPaletteTranslator.getBedrockBlockId(BlockPaletteTranslator.bedrockStateFromNBTMap(map.build()));
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+
+            if (storageReadIndex == 0) {
+                int index = 0;
+                for (int x = 0; x < 16; x++) {
+                    for (int z = 0; z < 16; z++) {
+                        for (int y = 0; y < 16; y++) {
+                            int paletteIndex = bitArray.get(index);
+                            int mcbeBlockId = sectionPalette[paletteIndex];
+                            if (mcbeBlockId != BlockPaletteTranslator.AIR_BEDROCK_BLOCK_ID) {
+
+                                BlockState blockState = BlockPaletteTranslator.RUNTIME_ID_TO_BLOCK_STATE.get(mcbeBlockId);
+
+                                chunkSections[sectionIndex].setBlockState(x, y, z, blockState);
+                            }
+                            index++;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Version one is the second newest chunk format by Bedrock.
+     * It is essentially the same as version eight, however there is only 1 layer per subchunk.
+     */
+    public static void networkDecodeVersionOne(ByteBuf byteBuf, ChunkSection chunkSection) {
+        networkDecodeVersionEight(byteBuf, new ChunkSection[]{chunkSection}, 0, (byte) 1);
+    }
+
+    /**
+     * Version zero is a chunk format used on PocketMine (3.0.0) and Nukkit (1.0) servers.
```

Comment:
Nukkit doesn't use this format - they use version 8.

Reply:
Woops, my bad. Completely thought they still used version 0.